### PR TITLE
Add experimental detrainment based on Pi groups

### PR DESCRIPTION
**Pull request by trontrytel**

### CODE REVIEW

**Comment by trontrytel** [comment]
This is ready for review. I'm trying to find a value of constant entrainment that does not crash the simulations but is larger than what I was showing during the group meeting. But the code itself will not change more.

File: src/TurbulenceConvection/closures/entr_detr.jl [hunk 1]
```diff
@@ -144,12 +145,89 @@ function compute_entr_detr!(
...
+                tke_en = aux_en.tke[k] # environment tke
+
+                w_up = w_up_c[k] # updraft vertical velocit at cell centers
+                w_en = w_en_c[k] # environment vertical velocity at cell centers
+
+                #w_d = get_wstar(bflux)
+                T_gm = aux_gm.T[k]
+
+                #Π_1 = z * (b_up - b_en) / ((w_up - w_en)^2 + w_d^2)
+                Π_2 =
```
Comment (costachris) [inline src/TurbulenceConvection/closures/entr_detr.jl:200]:
```suggestion
                Π_2 = (tke_gm - a_en * tke_en) / (tke_gm + eps(FT))
```

File: src/TurbulenceConvection/closures/entr_detr.jl [hunk 2]
```diff
@@ -144,12 +145,89 @@ function compute_entr_detr!(
...
+
+                a_up = aux_up[i].area[k] # updraft area fraction
+                #b_up = aux_up[i].buoy[k] # updraft buoyancy
+                RH_up = aux_up[i].RH[k] # updraft relative humidity
+
+                a_en = aux_en.area[k] # environment area fraction
+                #b_en = aux_en.buoy[k] # environment buoyancy
+                RH_en = aux_en.RH[k] # environment relative humidity
+                tke_en = aux_en.tke[k] # environment tke
+
```
Comment (costachris) [inline src/TurbulenceConvection/closures/entr_detr.jl:192]:
```suggestion
tke_gm = aux_gm.tke[k] # grid-mean tke
```

**Review by costachris — APPROVED** [review]

File: src/TurbulenceConvection/closures/entr_detr.jl (hunk 1 above)
Comment (costachris) [reply src/TurbulenceConvection/closures/entr_detr.jl:200]: Π_2 was changed to this some time ago (which is a variant of the longer Π_2 form we discussed before). All calibrations (including the one for detrainment below) are with this version of Π_2 suggested here, which matched TC.jl

**Pull request merged**